src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -364,6 +364,274 @@ CREATE NONCLUSTERED INDEX IX_ChangeFeed_StudyInstanceUid_SeriesInstanceUid_SopIn
     SopInstanceUid
 )
 
+/*************************************************************
+    Custom Tag Table
+    Stores added custom tags
+    TagPath is represented without any delimiters and each level takes 8 bytes
+    TagLevel can be 0, 1 or 2 to represent Instance, Series or Study level
+**************************************************************/
+CREATE TABLE dbo.CustomTag (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagPath                 VARCHAR(64)          NOT NULL,
+    TagVR                   VARCHAR(2)           NOT NULL,
+    TagLevel                TINYINT              NOT NULL
+)
```

Comment:
If we're creating clustered indices on the primary key, should we just create a primary key constraint instead?

```sql
    TagLevel TINYINT NOT NULL,
    CONSTRAINT PK_CustomTag_TagKey PRIMARY KEY CLUSTERED (TagKey)
```

Reply:
Hmm, that's a good point. We don't currently seem to follow that pattern, I'll find out why!

---
In reply to: [565686889](https://github.com/microsoft/dicom-server/pull/523#discussion_r565686889) [](ancestors = 565686889)